### PR TITLE
Refactor RBF mappings to have a common base class

### PR DESCRIPTION
**Pull request by DavidSCN**
## Main changes of this PR
Moves common content of the PETSc RBF mappings and the eigen RBF mappings to a common base class.

## Motivation and additional information
There are several functions duplicated in both methods.

## Author's checklist

* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I ran `make format` to ensure everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

**Comment by fsimonis** [comment]
Great effort!

I always wondered how much logic could be shifted into an rbf base class, which doesn't even depend on the actual basis function itself.
This could reduce compilation time by quite a lot.

**Comment by DavidSCN** [comment]
> which doesn't even depend on the actual basis function itself.

I also thought about this, but it requires some additional effort. In order to keep a reasonable PR size, I would go with the current solution.

File: src/mapping/RadialBasisFctMapping.hpp [hunk 1]
```diff
@@ -51,101 +51,58 @@ class RadialBasisFctMapping : public Mapping {
...
-  setDeadAxis(xDead, yDead, zDead);
 }
 
 template <typename RADIAL_BASIS_FUNCTION_T>
 void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
 {
   PRECICE_TRACE();
 
-  precice::utils::Event e("map.rbf.computeMapping.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
+  precice::utils::Event e("map.rbf.computeMapping.From" + this->input()->getName() + "To" + this->output()->getName(), precice::syncMode);
```
Comment (boris-martin) [inline src/mapping/RadialBasisFctMapping.hpp:90]: What is the added value of adding "this" ?

**Comment by boris-martin** [comment]
Looks great to me! I didn't check the whole implementation of each subclass but the subdivision makes sense to me. Just curious about all the added `this` who seem redundant to me.

File: src/mapping/RadialBasisFctMapping.hpp (hunk 1 above)
Comment (DavidSCN) [reply src/mapping/RadialBasisFctMapping.hpp:90]: The compiler forces me to explicitly refer to the base class member variables here, as the base class is now a class template with a template parameter depending on the instantiation of the derived class. If not put here, the compiler will not find the variable in the first place.

**Pull request merged**